src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -273,7 +273,12 @@ bool parser::parse_and_visit_statement(
       this->skip();
       this->check_body_after_label();
       goto parse_statement;
-
+      // "async export f()" is not valid. It should be "export async f()"
+    case token_type::kw_export: {
+      this->diag_reporter_->report(
+          diag_async_export_method{async_token.span()});
+      break;
```

Comment:
Shouldn't we parse the exported function (line 231)? If we don't, then the following will also (incorrectly) report that we're using `await` in a not-async function:
```javascript
async export function f() {
  await myPromise;  // await should be an operator here.
}
```